vsintegration/src/FSharp.Editor/Options/SettingsStore.fs

```diff
@@ -80,17 +72,18 @@ module Settings =
                 settingsManager.SetValueAsync(storageKeyOf settings, JsonConvert.SerializeObject settings, false) |> ignore
 
             // This is the point we retrieve the initial value and subscribe to watch for changes.
-            member _.Register defaultSettings =
-                defaultSettings |> updateFromStore |> setValue cache
-                let subset = storageKeyOf defaultSettings |> settingsManager.GetSubset
-                // This event is also raised when a setting change occurs in another VS instance, so we can keep everything in sync.
-                // Note: In experimental VS instance it only picks up local changes.
-                subset.add_SettingChangedAsync(
-                    PropertyChangedAsyncEventHandler( fun _ _ ->
-                        backgroundTask { cache |> getValue |> updateFromStore |> setValue cache } )
-                )
-
-    let CreateStore() : SettingsStore =          
+            member _.Register defaultSettings = register defaultSettings
+            member _.Get() = cache |> getValue
+
+    let Create () : ISettingsStore = 
+        let cache = Cache()
+
         match ServiceProvider.GlobalProvider.GetService(Guids.svsSettingsPersistenceManagerId) with
-        | :? ISettingsManager as settingsManager -> PersistentStore(settingsManager)
-        | _ -> TestStore()
\ No newline at end of file
+        | :? ISettingsManager as settingsManager -> PersistentStore(cache, settingsManager)
+        | _ ->
+            // When running without Visual Studio, i.e. unit tests, we use this in-memory store.
+            { new ISettingsStore with
```

Comment:
I actually liked having a name for this (TestStore), now I think an even better one would be InMemorySettingsStore. Otherwise - looks good :)